### DISCUSSION TITLE
Palette Previewer

### DISCUSSION THREAD

**Discussion by giscus** [aduros/wasm4#127] [Games]
# Palette Previewer

Build retro games using WebAssembly for a fantasy console

https://wasm4.org/play/palette-previewer

**Comment by thedavesims** [comment]
# Palette Preview
This is a simple cart that allows you to cycle a test image through a variety of 4-color palettes. [Source Code available here](https://github.com/thedavesims/WASM-4-Palette-Previewer). Up and Down to cycle through the palettes. Z or X to invert the palette. 

The palettes are taken from [Lospec](https://lospec.com/palette-list), a great source of inspiration for 4-color retro inspired palettes. The [palettes are available as an array of C structs](https://gist.github.com/thedavesims/b76acc46107ba82366f31106e47ee62f) and can be easily copied into your project, enabling you to try out a variety of different palettes.

You can then apply the palette using code similar to this example:
```c
void set_palette(int pal, int invert)
{
    if(invert)
    {
        PALETTE[0] = lospec_palettes[pal].palette[0];
        PALETTE[1] = lospec_palettes[pal].palette[1];
        PALETTE[2] = lospec_palettes[pal].palette[2];
        PALETTE[3] = lospec_palettes[pal].palette[3];
    } else 
    {
        PALETTE[3] = lospec_palettes[pal].palette[0];
        PALETTE[2] = lospec_palettes[pal].palette[1];
        PALETTE[1] = lospec_palettes[pal].palette[2];
        PALETTE[0] = lospec_palettes[pal].palette[3];
    }
}
```
When you have decided on your palette you can assign it directly [as in the documentation](https://wasm4.org/docs/guides/basic-drawing).
